src/adera/Machines/RCSController.cpp

```diff
@@ -117,27 +108,36 @@ float SysMachineRCSController::thruster_influence(Vector3 posOset, Vector3 direc
 
 void SysMachineRCSController::update_controls(ActiveScene& rScene)
 {
-    auto view = rScene.get_registry().view<MachineRCSController, ACompTransform>();
+    auto view = rScene.get_registry().view<MachineRCSController>();
 
     for (ActiveEnt ent : view)
     {
         auto& machine = view.get<MachineRCSController>(ent);
-        auto& transform = view.get<ACompTransform>(ent);
 
         using wiretype::AttitudeControl;
         using Magnum::Vector3;
 
+        auto physComps = SysPhysics_t::try_get_or_find_rigidbody_parent(
+            rScene, ent, machine.m_rigidBody);
+        if (!physComps.first || !physComps.second) { continue; }
+
+        ACompRigidBody_t *compRb = physComps.first;
+        ACompTransform *compTf = physComps.second;
+
         if (WireData *gimbal = machine.m_wiCommandOrient.connected_value())
         {
             AttitudeControl *attCtrl = std::get_if<AttitudeControl>(gimbal);
 
+            // TODO: cache this
+            Magnum::Matrix4 relTransform{};
+            SysPhysics_t::find_rigidbody_ancestor(rScene, ent, &relTransform);
```

Comment:
I'd rather the function just return the Matrix4, instead of using a reference-parameter.

Reply:
The optional parameter allows the linear algebra to be skipped if the transformation isn't requested. I'll try to think up a cleaner solution though.

Reply:
Two different functions.